shared/gh/js/views/gh.datepicker.js

```diff
@@ -222,11 +208,167 @@ define(['gh.core', 'moment', 'clickover', 'jquery-datepicker'], function(gh, mom
+        $('.popover #gh-datepicker').datepicker({
+            'dateFormat': "yy-mm-dd",
+            'dayNamesMin': ['S','M','T','W','T','F','S'],
+            'firstDay': 4,
+            'nextText': '<i class="fa fa-chevron-right"></i>',
+            'prevText': '<i class="fa fa-chevron-left"></i>',
+            'showOtherMonths': true,
+            'selectOtherMonths': true,
+            'onSelect': onDatepickerChange
+        });
+    };
+
+    /**
+     * Navigate to a specific date in the calendar
+     *
+     * @param  {String}    date    The date that needs to be displayed in the calendar
+     * @private
+     */
+    var setDate = function(date) {
+        $('.popover #gh-datepicker').datepicker('setDate', date);
+    };
+
+    /**
+     * Set the original date and time
+     *
+     * @private
+     */
+    var setCalendarDate = function() {
+        // Set the current date
+        var dayValue = moment($trigger.attr('data-start')).utc().format('YYYY-MM-DD');
+        setDate(dayValue);
+
+        // Set the correct select box values
+        var startDate = gh.api.utilAPI.convertUnixDatetoISODate(gh.api.utilAPI.fixDateToGMT($trigger.attr('data-start')));
+        var endDate = gh.api.utilAPI.convertUnixDatetoISODate(gh.api.utilAPI.fixDateToGMT($trigger.attr('data-end')));
+        var week = gh.api.utilAPI.getWeekInTerm(startDate);
+        var day = moment(startDate).day();
+
+        $('.popover #gh-module-week').val(week);
+        $('.popover #gh-module-day').val(day);
+
+        $('.popover #gh-module-from-hour').val(moment(startDate).utc().format('HH'));
+        $('.popover #gh-module-from-minutes').val(moment(startDate).utc().format('mm'));
+
+        $('.popover #gh-module-to-hour').val(moment(endDate).utc().format('HH'));
+        $('.popover #gh-module-to-minutes').val(moment(endDate).utc().format('mm'));
+    };
+
+    /**
+     * Function that is executed when the datepicker value was changed
+     *
+     * @private
+     */
+    var onDatepickerChange = function() {
+        updateDateComponents($('#gh-datepicker'));
+        validateEntry();
+    };
+
+
+    ////////////////////
+    //  FORM CHANGES  //
+    ////////////////////
+
+
+    /**
+     * Function that is executed when the week has been changed
+     *
+     * @private
+     */
+    var onWeekChange = function() {
+        updateDateComponents($('#gh-module-week'));
+        validateEntry();
+    };
+
+    /**
+     * Function that is exectued when the day has been changed
+     *
+     * @private
+     */
+    var onDayChange = function() {
+        updateDateComponents($('#gh-module-day'));
+        validateEntry();
+    };
+
+
     ///////////////
     //  BINDING  //
     ///////////////
 
     /**
+     * Update the date components
+     *
+     * @param  {Object}    trigger    jQuery object representing the component that invoked the change
+     * @private
+     */
+    var updateDateComponents = function(trigger) {
```

Comment:
Should we pass in the ID of the selector instead of the jQuery object? It'd be a bit cleaner than doing `$(trigger).selector` later on.